src/components/Layout/Main/Main.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { useAllCountries } from '../../../hooks/useCountries.js';
+import CountryCard from '../../CountryCard/CountryCard.js';
+
+import background from '../../../background.jpg';
+import './Main.css';
+
+export default function Main() {
+  const [order, setOrder] = useState('');
+  const [searchByName, setSearchByName] = useState('');
+  const [loadState, setLoadState] = useState('loading');
+
+  const { countries, error } = useAllCountries(order, searchByName, setLoadState);
+
+  const filteredCountries = countries.filter((countriesArray) => countriesArray);
+
+  const continents = [...new Set(filteredCountries.map(({ continent }) => continent))];
+
+  const filteredContinents = continents.map((continent) => {
+    if (continent === null) {
+      continent = 'All';
+      return continent;
+    } else {
+      return continent;
+    }
+  });
+
+  const [selectedContinent, setSelectedContinent] = useState('All');
+
+  const handleContinentChange = (e) => {
+    setSelectedContinent(e.target.value);
+  };
+
+  return (
+    <main
+      style={{
+        backgroundImage: `url(${background})`,
+        // backgroundSize: 'cover',
+        // backgroundRepeat: 'no-repeat',
+        // backgroundPosition: 'center',
+      }}
+    >
+      <div className="search-bar-container">

```

Comment:
This would be a good spot to turn this into its own component, since you have multiple search functions here you could create a `<Nav/>` component with all of your search functionality and this would make this file significantly shorter.